src/Shared/CoreCLRAssemblyLoader.cs

```diff
@@ -104,7 +104,9 @@ private Assembly TryResolveAssembly(AssemblyLoadContext context, AssemblyName as
                 {
                     foreach (var extension in _extensions)
                     {
-                        var candidatePath = Path.Combine(dependencyPath, $"{assemblyName.Name}.{extension}");
+                        var candidatePath = Path.Combine(dependencyPath,
+                            assemblyName.CultureName ?? string.Empty,
```

Comment:
Separate question about line after this: how can we get this far with the assembly already loaded? It looks like it would fail to reuse the already loaded assembly in that case.